From ecb4e533f38dbd5a2ffaa6687ad715fa8b011a2e Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Tue, 21 Jun 2022 14:13:32 +0200
Subject: [PATCH 0263/1141] arm64: dts: st: add power syscon on stm32mp251

The Cortex-M33 management needs this node as the
PWR_CPU2D2SR register is used to determine the Cortex-M33 state.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I7430051f1ad071737ec52a8c0a100dd0ebacada7
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/310432
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -1959,6 +1959,11 @@
 			feature-domains = <&rifsc 156>;
 		};
 
+		pwr: syscon@44210000 {
+			compatible = "st,stm32mp25-pwr", "syscon";
+			reg = <0x44210000 0x0400>;
+		};
+
 		exti1: interrupt-controller@44220000 {
 			compatible = "st,stm32mp1-exti", "syscon";
 			interrupt-controller;
@@ -2312,6 +2317,7 @@
 			resets = <&scmi_reset RST_SCMI_C2_R>,
 				 <&scmi_reset RST_SCMI_C2_HOLDBOOT_R>;
 			reset-names = "mcu_rst", "hold_boot";
+			st,syscfg-cm-state = <&pwr 0x204 0x0000000c>;
 			interrupt-parent = <&intc>;
 			interrupts = <GIC_SPI 4 IRQ_TYPE_LEVEL_HIGH>;
 
